File: scripts/update-image.mjs
#!/usr/bin/env node
import fs from 'node:fs';
import path from 'node:path';
import { fileURLToPath } from 'node:url';

const [, , environment, imageName, newName, newTag] = process.argv;
if (![environment, imageName, newName, newTag].every(Boolean)) {
  throw new Error('Uso: update-image.mjs <ambiente> <imagen> <repositorio> <sha>');
}
if (!/^[a-f0-9]{40}$/.test(newTag)) {
  throw new Error('La etiqueta debe ser un SHA completo de 40 caracteres');
}

const repositoryRoot = path.resolve(path.dirname(fileURLToPath(import.meta.url)), '..');
const manifestPath = path.join(repositoryRoot, 'overlays', environment, 'kustomization.yaml');
const lines = fs.readFileSync(manifestPath, 'utf8').split(/\r?\n/);
const imageLine = lines.findIndex((line) => line.trim() === `- name: ${imageName}`);
if (imageLine < 0 || imageLine + 2 >= lines.length) {
  throw new Error(`No se encontró la imagen ${imageName} en ${manifestPath}`);
}
lines[imageLine + 1] = `    newName: ${newName}`;
lines[imageLine + 2] = `    newTag: ${newTag}`;
fs.writeFileSync(manifestPath, `${lines.join('\n').replace(/\n+$/, '')}\n`, 'utf8');

File: scripts/update-image.py
#!/usr/bin/env python3
import pathlib
import sys


def update_image(environment: str, image_name: str, new_name: str, new_tag: str) -> None:
    path = pathlib.Path(__file__).resolve().parents[1] / "overlays" / environment / "kustomization.yaml"
    lines = path.read_text(encoding="utf-8").splitlines()
    updated = False
    for index, line in enumerate(lines):
        if line.strip() == f"- name: {image_name}":
            if index + 2 >= len(lines):
                break
            lines[index + 1] = f"    newName: {new_name}"
            lines[index + 2] = f"    newTag: {new_tag}"
            updated = True
            break
    if not updated:
        raise SystemExit(f"No se encontró la imagen {image_name} en {path}")
    path.write_text("\n".join(lines) + "\n", encoding="utf-8")


if __name__ == "__main__":
    if len(sys.argv) != 5:
        raise SystemExit("Uso: update-image.py <ambiente> <imagen> <repositorio> <sha>")
    update_image(*sys.argv[1:])
